homeassistant/components/axis/config_flow.py

```diff
@@ -65,8 +65,15 @@ def __init__(self):
 
                 self.serial_number = device.vapix.params.system_serialnumber
 
+                for entry in self.hass.config_entries.async_entries(DOMAIN):
+                    if self.serial_number == entry.unique_id:
+                        return self._update_entry(
+                            entry,
+                            host=user_input[CONF_HOST],
+                            port=user_input[CONF_PORT],
+                        )
+
                 await self.async_set_unique_id(self.serial_number)
```

Comment:
This function return the already configured entry that you lookup above.

Reply:
Yes but it also sets the unique id attribute

Reply:
Yes but you already check above if the unique id is the same as the serial number. If it is, you update that entry with just that unique id. Using the returned entry to update will be the same thing as you lookup above.

Reply:
This call is only to assign the unique id to the new entry. If it is an existing entry it will have been aborted already.

Reply:
Damn, I really didn't read what you meant. Now I understand. Thanks for the feedback!